scripts/common/proxy.sh

```diff
@@ -48,6 +48,45 @@ function proxy_bootstrap() {
     fi
 }
 
+# check_proxy_config tries to check if is possible connect with the registry
+# the following code will check if the proxy is invalid by doing a check as following
+# ~$ sudo HTTP_PROXY=http://77.89.35.50:8080 HTTPS_PROXY=http://77.89.35.50:8080 ctr image pull docker.io/library/nginx:latest
+# docker.io/library/nginx:latest: resolving      |--------------------------------------|
+# elapsed: 29.9s                  total:   0.0 B (0.0 B/s)
+# INFO[0030] trying next host                              error="failed to do request: Head \"https://registry-1.docker.io/v2/library/nginx/manifests/latest\": proxyconnect tcp: dial tcp 77.89.35.50:8080: i/o timeout" host=registry-1.docker.io
+# ctr: failed to resolve reference "docker.io/library/nginx:latest": failed to do request: Head "https://registry-1.docker.io/v2/library/nginx/manifests/latest": proxyconnect tcp: dial tcp 77.89.35.50:8080: i/o timeout
+function check_proxy_config() {
+    if [ -z "$CONTAINERD_VERSION" ]; then
+       return
+    fi
+
+    logStep "Checking proxy configuration with Containerd"
+
+    # Echo containerd Proxy config:
+    local proxy_config_file="/etc/systemd/system/containerd.service.d/http-proxy.conf"
+    if [ ! -f "$proxy_config_file" ]; then
+        log "Skipping test. No HTTP proxy configuration found."
+        return
+    fi
+
+    echo ""
+    log "Proxy config:"
+    grep -v -e '^\[Service\]' -e '^# Generated by kURL' "$proxy_config_file"
+    echo ""
+
+    if ! response=$(crictl pull test/invalid/image:latest 2>&1) || [[ $response =~ .*"proxy".* ]]; then
```

Comment:
Have you tested this with a valid proxy? Wont the pull fail since the image is invalid? Will it not enter this conditional in that case?